CONTRIBUTING.md

```diff
@@ -22,19 +21,47 @@ See the [XAITK-Saliency README] file for additional information.
 
         $ git push origin HEAD -u
 
-5.  Visit the Kitware XAITK-Saliency Github, browse to the "Pull requests" tab
+5.  Visit the Kitware XAITK-Saliency GitHub, browse to the "Pull requests" tab
     and click on the "New pull request" button in the upper-right.
     Click on the "compare across forks" link, browse to your fork and browse to
     the topic branch to submit for the pull request.
     Finally, click the "Create pull request" button to create the request.
 
 
-XAITK-Saliency uses GitHub for code review and Github Actions for continuous
-testing as new pull requests are made.
-All checks/tests must pass before a PR can be merged.
+XAITK-Saliency uses GitHub for code review and GitHub Actions for continuous
+testing.
+New pull requests trigger CI when the merge target is the `master` branch.
+All checks/tests must pass before a PR can be merged by an individual with the
+appropriate permissions.
+
+We use Sphinx for manual and automatic API [documentation](docs).
+
 
-Sphinx is used for manual and automatic API [documentation].
+## Class Naming Philosophy
+A class that embodies performing a central action should be named following the
+"Verb + Noun" style.
+For example, a name that follows this might be "PerformOperation" as opposed to
+"OperationPerformer".
+This is especially important for user-facing classes like interfaces and
+utilities.
 
+Some concrete examples as can be found in this repository are:
+* [`PerturbImage`](xaitk_saliency/interfaces/perturb_image.py)
+    * action: `Perturn`
+    * noun: `Image`
+* [`GenerateImageClassifierBlackboxSaliency`](xaitk_saliency/interfaces/vis_sal_image_classifier_blackbox.py)
+    * action: `Generate`
+    * noun: `ImageClassifierBlackboxSaliency`
 
-[XAITK-Saliency README]: README.md
-[documentation]: docs/
+A special case for class naming can be made for **concrete implementation**
+of interface classes, as they repeat the verb-noun semantic introduced by the
+parent interface class(es).
+These sub-classes may not always be required to strictly follow the verb-noun
+pattern as doing so will often create overly repetitive and lengthy class
+names.
+The repetitiveness is a non-trivial consideration due to the interplay with
```

Comment:
Replaced use of "action" with "verb" to be consistent.

Reply:
I'm not sure how to incorporate this into the first paragraph meaningfully because this sentence incorporates both plugins and the implementation "exception."